### PR TITLE
Repo controlled build go version

### PR DESCRIPTION
**Pull request by xdu31**
**What this PR does / why we need it**:

Follow [KEP-3744](https://github.com/kubernetes/enhancements/issues/3744) to make `aws-iam-authenticator` to stay on supported Go versions.

This PR follows https://github.com/kubernetes/kubernetes/pull/114660 and https://github.com/kubernetes/kubernetes/pull/115377 for repo-controlled go version to use in CI. 

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

### CODE REVIEW

File: .github/workflows/deps.yml [hunk 1]
```diff
@@ -19,18 +19,3 @@ jobs:
         uses: actions/checkout@v4
         with:
           show-progress: false
-      - id: govulncheck
```
Comment (micahhausler) [inline .github/workflows/deps.yml:22]: Why remove govunlcheck?

Comment (xdu31) [reply .github/workflows/deps.yml:22]: > Why remove govunlcheck?

Hi @micahhausler I need to remove it temporarily to make the premerge test move forward, I will add it back when your PR https://github.com/kubernetes-sigs/aws-iam-authenticator/pull/638 to resolve the govunlcheck is merged.

Comment (xdu31) [reply .github/workflows/deps.yml]: Hi @micahhausler I have rebased the PR with go 1.21.5 version and added back the govunlcheck, could you take another look?

**Review by jyotimahapatra — APPROVED** [review]

**Pull request merged**